pytest_factoryboy/fixture.py

```diff
@@ -163,14 +141,70 @@ def register_(factory_class: F) -> F:
         )
     )
 
-    generated_module = make_fixture_model_module(model_name, fixture_defs)
 
-    for fixture_def in fixture_defs:
-        exported_name = fixture_def.name
-        fixture_function = getattr(generated_module, exported_name)
-        inject_into_caller(exported_name, fixture_function, _caller_locals)
+def make_attribute_fixturedef(
+    overrides: Mapping[str, Any],
+    attr: str,
+    model_name: str,
+    factory_value: Any,
+    factory_class: FactoryType,
+    related: list[str],
+) -> FixtureDef:
+    attr_name = SEPARATOR.join((model_name, attr))
+    if attr in overrides:
+        # Attribute is overridden in the register(...) call, let's just take the value and done.
+        value = overrides[attr]
+
+        return FixtureDef(
+            name=attr_name,
+            function_name="attr_fixture",
+            function_kwargs={"value": value},
+            deps=value.args if isinstance(value, LazyFixture) else [],
+        )
 
-    return factory_class
+    # Attribute is not overridden, so we must check what the factory value is and handle each case
+    if isinstance(factory_value, (factory.SubFactory, factory.RelatedFactory)):
+        subfactory_class = factory_value.get_factory()
+        subfactory_deps = get_deps(subfactory_class, factory_class)
+
+        args = list(subfactory_deps)
+        if isinstance(factory_value, factory.RelatedFactory):
+            related_model = get_model_name(subfactory_class)
+            args.append(related_model)
+            related.append(related_model)
+            related.append(attr_name)
+            related.extend(subfactory_deps)
+
+        if isinstance(factory_value, factory.SubFactory):
+            args.append(inflection.underscore(subfactory_class._meta.model.__name__))
+
+        return FixtureDef(
+            name=attr_name,
+            function_name="subfactory_fixture",
+            function_kwargs={"factory_class": subfactory_class},
+            deps=args,
+        )
+
+    if isinstance(factory_value, factory.PostGeneration):
+        value = None
+    elif isinstance(factory_value, factory.PostGenerationMethodCall):
+        value = factory_value.method_arg
+    else:
+        value = factory_value
+
+    # Do we want to allow to specify LazyFixture in the Factory class definition? Probably not
+    # TODO: if so, remove this
```

Comment:
I agree that it's a wrong usage of `LazyFixture` :+1:

Reply:
Although I can imagine that it's handy to define an attribute as a LazyFixture if your factory is really only usable in a pytest context. Maybe we should support this (= add test coverage), it's literally only this if-else that we must keep on doing.

Reply:
Yeap, I agree, however imho it's very unclean because it will be impossible to use factory class as a regular factoryboy factory, but when the whole development team decides on such an approach in their projects that's fine.